### PR TITLE
[main-aqm] feat: add initial docstrings and typing support to smoke/dust ush scripts

### PR DESCRIPTION
**Pull request by benkozi**
* Adds initial docstrings to all smoke and dust Python functions in ush scripts.
* Adds `typing` support to all arguments and return values.
  - Removed some unused arguments from function signatures.
  - Some small code clean-up to fix basic type checking.

To run `mypy`, I used the following command:

> mypy --disallow-untyped-defs add_smoke.py interp_tools.py fire_emiss_tools.py HWP_tools.py generate_fire_emissions.py

`mypy` could eventually be added to the CI pipelines to ensure there is no new type drift.

Output is as follows:

```
(srw_app) root@91d77a164633:/opt/srw/ush# mypy --disallow-untyped-defs add_smoke.py interp_tools.py fire_emiss_tools.py HWP_tools.py generate_fire_emissions.py 
interp_tools.py:11: error: Skipping analyzing "netCDF4": module is installed, but missing library stubs or py.typed marker  [import-untyped]
interp_tools.py:11: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
interp_tools.py:17: error: Skipping analyzing "esmpy": module is installed, but missing library stubs or py.typed marker  [import-untyped]
interp_tools.py:20: error: Cannot find implementation or library stub for module named "ESMF"  [import-not-found]
fire_emiss_tools.py:9: error: Skipping analyzing "netCDF4": module is installed, but missing library stubs or py.typed marker  [import-untyped]
fire_emiss_tools.py:153: error: No overload variant of "__sub__" of "datetime" matches argument type "int"  [operator]
fire_emiss_tools.py:153: note: Possible overload variants:
fire_emiss_tools.py:153: note:     def __sub__(self, datetime, /) -> timedelta
fire_emiss_tools.py:153: note:     def __sub__(self, timedelta, /) -> datetime
fire_emiss_tools.py:153: note: Right operand is of type "datetime | int"
Found 5 errors in 2 files (checked 5 source files)
```

### Type of change

- [x] New feature (non-breaking change which adds functionality)

## TESTS CONDUCTED: 

- [ ] derecho.intel
- [ ] gaea.intel
- [ ] hera.gnu
- [ ] hera.intel
- [ ] hercules.intel
- [ ] jet.intel
- [x] orion.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

### CODE REVIEW

File: ush/interp_tools.py [hunk 1]
```diff
@@ -88,7 +124,7 @@ def check_for_raw_rave(RAVE, intp_non_avail_hours, intp_avail_hours):
             rave_nonavail_hours_test.append(date)
         else:
             print(f'Raw RAVE available for interpolation {matching_files}')
-            rave_avail.append(matching_files)
+            rave_avail.extend(matching_files)
```
Comment (BenKoziol-NOAA) [inline ush/interp_tools.py:127]: A downstream function expects `List[str]`, but appending will create a `List[List[str]]`.

**Comment by chan-hoo** [comment]
@BenKoziol-NOAA, the 'smoke_dust' task fails with:
```
Traceback (most recent call last):
  File "/scratch2/NAGAPE/epic/Chan-hoo.Jeon/pr_main_aqm/ufs-srweather-app/ush/generate_fire_emissions.py", line 120, in <module>
    generate_emiss_workflow(sys.argv[1], sys.argv[2], sys.argv[3], sys.argv[4], sys.argv[5], sys.argv[6],sys.argv[7])
  File "/scratch2/NAGAPE/epic/Chan-hoo.Jeon/pr_main_aqm/ufs-srweather-app/ush/generate_fire_emissions.py", line 95, in generate_emiss_workflow
    femmi_tools.produce_emiss_24hr_file(frp_avg_reshaped, nwges_dir, current_day, tgt_latt, tgt_lont, ebb_total_reshaped, cols, rows)
  File "/scratch2/NAGAPE/epic/Chan-hoo.Jeon/pr_main_aqm/ufs-srweather-app/ush/fire_emiss_tools.py", line 192, in produce_emiss_24hr_file
    i_tools.Store_latlon_by_Level(fout, 'geolat', tgt_latt, 'cell center latitude', 'degrees_north', '2D', '-9999.f', '1.f')
TypeError: Store_latlon_by_Level() takes 6 positional arguments but 8 were given
```

**Comment by benkozi** [comment]
Oops, I must have gotten my wires crossed when testing. I pushed a fix and ran the task successfully on Orion. Hopefully you'll have similar results. Thanks for catching.

**Comment by chan-hoo** [comment]
It looks working well now. Approving.

**Review by chan-hoo — APPROVED** [review]

**Pull request merged**